Don't try find_library tests if the system doesn't satisfy the assumptions

yaxxie opened the pull request:
Recent M1 (cross architecture libs) issues and a weird docker image highlight a few circumstances where the assumed-to-be-present libs for the `find_library` tests aren't always present (in the expected forms, i.e. `libname.system_extension`)

To stop this test from failing and causing issues with users, if no such lib is found, the system is declared "broken" and these tests will subsequently be skipped as "broken", which doesn't cause alarming test failure messages.

danielsoutar commented on test/basic/test_lightgbm.jl:
```diff
@@ -44,37 +61,49 @@ end
-        cp(settings["ref_lib_lightgbm_path"], settings["lib_not_on_sys_fixture_path"]) # fake file copied from lightgbm
+        settings, isborked = setup_env()
+        if !isborked
+            cp(settings["linkable_path"], settings["lib_not_on_sys_fixture_path"], force=true, follow_symlinks=true)
```
This and its counterpart in the test below seems to be the offending lines for the macOS CI failures:
```
 IOError: open("/usr/lib/libSystem.B.dylib", 0, 0): no such file or directory (ENOENT)
```

Looking at the Apple developer forums, they mention [`dlopen` code potentially failing if file existence is checked](https://developer.apple.com/forums/thread/655588).

yaxxie commented:
@danielsoutar I've reverted these tests to the basic form. Now it will detect if its a broken system and just show these tests as "broken", which won't cause alarming failures. In addition, I refactored it to run setup only once, and removed a redundant test that wasn't checking anything new.
